### ISSUE TITLE
Windows native 7.1 to Jack

### ISSUE THREAD

**Issue by marijncin** [jackaudio/jack2#686]
Hi, Not an issue here, just a question. 

Windows 10

I am using VB Hifi cable asiobridge to send my windows native audio to Jack. I would like to do this for more then two channels. VB cable won't let me. 
Does anybody know of a good way to port windows native apps. (like say for example a 5.1 videogame) to jack? 

Greeting, Marijn

**Comment by falkTX** [comment]
the best way to port applications is to actually *port* them.
that is, add jack support. the API is quite simple all things considered.

**Comment by marijncin** [comment]
@falkTX Thanks, Yes I understand. For applications I am building, that is indeed the best way. But how would that work for an external application made by someone else? For example Let's say I want to play far cry 6, or some game on 5.1 and send the output to Jack? Most apps like this only work with Windows Native.  I'm looking for a tool to port windows native.

**Comment by falkTX** [comment]
jack-router is a possibility. current jack2 release do not ship with it, but might in the future (made optional) since now we have working binaries again.
but even then, jack-router is for applications that talk ASIO, and stuff like games usually does not support that.

so I think the only solution here is some sort of virtual soundcard and a bridge to bring that soundcard into the jack graph.
this will have at least 1 cycle of audio latency though.

**Comment by marijncin** [comment]
> jack-router is a possibility. current jack2 release do not ship with it, but might in the future (made optional) since now we have working binaries again.
> but even then, jack-router is for applications that talk ASIO, and stuff like games usually does not support that.
> 
> so I think the only solution here is some sort of virtual soundcard and a bridge to bring that soundcard into the jack graph.
> this will have at least 1 cycle of audio latency though.

Exactly! "some sort of virtual soundcard" is what I am looking for! I tried VB Cable, but only let's me do stereo.

**Comment by falkTX** [comment]
or as a last resort, use an actual soundcard that can provide 8 output channels, and feed that into your main card inputs ?

**Comment by marijncin** [comment]
> or as a last resort, use an actual soundcard that can provide 8 output channels, and feed that into your main card inputs ?

Yes, That would work. However, there must be a better way. I came across something called Synchronous audio router, but it does not work anymore unfortunately. (at least, I don't get it to work).

**Comment by mr-berndt** [comment]
Sorry for chiming in late but I might shed some insight.

I am using VB HiFi Cable wit 8 channels out of kodi, feeding them into jack and sending them to linux. This works, I use it since years, so it might be worth checking again.

Another option is to get a MiniDSP USB Streay<mer and configure it for ADAT and use fibreoptics cable to transport your 8 channels. Downside is, it will only be 48 kHz @ 16 bits. Dunno if this is enough for you.

**Comment by marijncin** [comment]
> Sorry for chiming in late but I might shed some insight.
> 
> I am using VB HiFi Cable wit 8 channels out of kodi, feeding them into jack and sending them to linux. This works, I use it since years, so it might be worth checking again.
> 
> Another option is to get a MiniDSP USB Streay<mer and configure it for ADAT and use fibreoptics cable to transport your 8 channels. Downside is, it will only be 48 kHz @ 16 bits. Dunno if this is enough for you.

Thank you for your reply. In the meantime I have managed to get Voicemeeter working with Jack. This works great, thanks.

**Comment by marijncin** [comment]
> Sorry for chiming in late but I might shed some insight.
> 
> I am using VB HiFi Cable wit 8 channels out of kodi, feeding them into jack and sending them to linux. This works, I use it since years, so it might be worth checking again.
> 
> Another option is to get a MiniDSP USB Streay<mer and configure it for ADAT and use fibreoptics cable to transport your 8 channels. Downside is, it will only be 48 kHz @ 16 bits. Dunno if this is enough for you.

Since you seem to be using Jack on Windows, might I ask you some burning questions? I could really use some help with my project.

**Comment by mr-berndt** [comment]
Didn't know voicemeteer, will check it out. Cool.

You can ask me, but all I use it for is sending the audio to the Linux box, so I might not be the right guy to ask. But please, fire away..

**Comment by mr-berndt** [comment]
But tell me one thing about voicemeteer: Is it restoring a useable connection, if the server side of jack disappears and comes back on? HiFi cable needs a click, meaning a keyboard, which is actually not connected to the video-player..

**Comment by marijncin** [comment]
> Didn't know voicemeteer, will check it out. Cool.
> 
> You can ask me, but all I use it for is sending the audio to the Linux box, so I might not be the right guy to ask. But please, fire away..

Voicemeeter is also from VB cable, it has a bit more features then HIFI cable, and does all the same things. 
---------------------------------------------------------------------------------
Currently, my problem is this: https://github.com/jackaudio/jack2/issues/719 
I can't get JackRouter to work Properly on windows. Strange thing is: On some PC's it works well, on others it does not.

**Comment by mr-berndt** [comment]
I would not really know enough to help with your issue, sorry..

**Comment by marijncin** [comment]
> But tell me one thing about voicemeteer: Is it restoring a useable connection, if the server side of jack disappears and comes back on? HiFi cable needs a click, meaning a keyboard, which is actually not connected to the video-player..

yes, it will.

**Comment by marijncin** [comment]
> But tell me one thing about voicemeteer: Is it restoring a useable connection, if the server side of jack disappears and comes back on? HiFi cable needs a click, meaning a keyboard, which is actually not connected to the video-player..

Don't think so.
